quota/etcd/README.md

```diff
@@ -0,0 +1,246 @@
+Assuming an expected sequencing performance of 50 QPS, the max_tokens specified
+implies a backlog of 4h.
+
+```bash
+curl \
+  -d '@-' \
+  -s \
+  -H 'Content-Type: application/json' \
+  -X POST \
+  "localhost:8091/v1beta1/quotas/global/write/config" <<EOF
+{
+  "name": "quotas/global/write/config",
+  "config": {
+    "state": "ENABLED",
+    "max_tokens": 288000,
+    "sequencing_based": {
+    }
+  }
+}
+EOF
+```
+
+To list all configured quotas, do:
+
+```bash
+curl localhost:8091/v1beta1/quotas?view=FULL
+```
+
+Quotas may be retrieved individually or via a series of filters, updated and
+deleted through the REST API as well. See
+[quotapb.proto](https://github.com/google/trillian/blob/master/quota/etcd/quotapb/quotapb.proto)
+for an in-depth description of entities and available methods.
+
+### Maintenance and token exhaustion
+
+During regular system operation, no quota-related maintenance should be
+required, as the system should generate at least as many tokens as it spends.
+
+If a token outage happens there are a few built-in mechanisms that allow manual
+intervention. The question of whether intervention is needed, though, is an
+important one and should be answered before any attempts to bypass the system.
+For example:
+
+* Is logsigner working properly and able to keep with the current demand?
+* Is there a spike in requests that may justify the current token outage?
+
+In genuine outages, it may be beneficial to let the quota system deny requests
+until regular operation is resumed.
+
+That said, the sections below describe a few alternatives that may used to deal
+with token outages. All examples use **global/read** as the quota in question;
+substitute the name as appropriate.
+
+#### Resetting quotas
+
+Resetting a quota restores its current token count to the configured
+**max_tokens** value.
+
+```bash
+curl -X PATCH localhost:8091/v1beta1/quotas/global/read/config?reset_quota=true
+```
+
+#### Disabling quotas
+
+Disables a quota makes it inactive, effective immediately. Disabled quotas may
+be enabled again with a similar update (changing "DISABLED" to "ENABLED").
+
+```bash
+curl \
+  -d '@-' \
+  -s \
+  -H 'Content-Type: application/json' \
+  -X PATCH \
+  "localhost:8091/v1beta1/quotas/global/read/config" <<EOF
+{
+  "config": {
+    "state": "DISABLED"
+  },
+  "update_mask": ["state"]
+}
+EOF
+```
+
+#### Deleting quotas
+
+Permanently deletes a quota. Consider disabling for a temporary solution.
+
+```bash
+curl -X DELETE localhost:8091/v1beta1/quotas/global/read/config
+```
+
+### Flags
+
+The following flags apply to etcd quotas:
+
+* [--quota_dry_run](https://github.com/google/trillian/blob/3cf59cdfd0/server/trillian_log_server/main.go#L61)
+  (log and map servers)
+* [--quota_increase_factor](https://github.com/google/trillian/blob/3cf59cdfd0/server/trillian_log_signer/main.go#L60)
+  (logsigner)
+
+`--quota_dry_run`, when set to true, stops quota errors from blocking requests.
```

Comment:
errors and depletion?

Reply:
Did you mean *'s/errors/depletion'*? (That's what I did.)